client/api_client.go

```diff
@@ -147,6 +147,9 @@ type ApiClientInterface interface {
 	ProjectBudget(projectId string) (*ProjectBudget, error)
 	ProjectBudgetUpdate(projectId string, payload *ProjectBudgetUpdatePayload) (*ProjectBudget, error)
 	ProjectBudgetDelete(projectId string) error
+	EnableUpdateEnvironmentDiscovery(projectId string, payload *EnvironmentDiscoveryPutPayload) (*EnvironmentDiscoveryPayload, error)
```

Comment:
I'd rename it to `PutEnvironmentDiscovery`, or just `UpdateEnvironmentDiscovery`